packages/my-queries-storage/src/compass-query-storage.ts

```diff
@@ -12,69 +12,119 @@ export type QueryStorageOptions = {
+    return this.backend.readOne(id);
+  }
+
+  async updateAttributes(id: string, patch: Partial<TData>): Promise<TData> {
+    return this.backend.updateAttributes(id, patch);
+  }
+
+  async delete(id: string): Promise<boolean> {
+    return this.backend.delete(id);
+  }
+
+  abstract saveQuery(data: Omit<TData, '_id' | '_lastExecuted'>): Promise<void>;
 }
 
-export abstract class CompassQueryStorage<
+// UserDataBackend preserves all major functionality and adapts UserData to the QueryStorageBackend interface
+export class UserDataBackend<
   TSchema extends z.Schema,
-  TData extends z.output<TSchema> = z.output<TSchema>
+  TData extends z.output<TSchema> & QueryDocumentBase
 > implements QueryStorageBackend<TData>
 {
-  protected readonly userData: UserData<TSchema>;
-  constructor(
-    schemaValidator: TSchema,
-    protected readonly folder: string,
-    protected readonly options: QueryStorageOptions
-  ) {
-    this.userData = new UserData(schemaValidator, {
-      subdir: folder,
-      basePath: options.basepath,
-      serialize: (content) => EJSON.stringify(content, undefined, 2),
-      deserialize: (content) => EJSON.parse(content),
-    });
+  private userData: UserData<TSchema>;
+
+  constructor(userData: UserData<TSchema>) {
+    this.userData = userData;
   }
 
-  async loadAll(namespace?: string): Promise<TData[]> {
+  async loadAll(): Promise<TData[]> {
     try {
       const { data } = await this.userData.readAll();
-      const sortedData = data
-        .sort((a, b) => {
-          return b._lastExecuted.getTime() - a._lastExecuted.getTime();
-        })
-        .filter((x) => !namespace || x._ns === namespace);
-      return sortedData;
+      return data as TData[];
     } catch {
       return [];
     }
   }
 
-  async updateAttributes(id: string, data: Partial<TData>): Promise<TData> {
-    await this.userData.write(id, {
-      ...((await this.userData.readOne(id)) ?? {}),
-      ...data,
-    });
-    return await this.userData.readOne(id);
+  async readOne(id: string): Promise<TData | undefined> {
+    try {
+      const result = await this.userData.readOne(id);
+      return result as TData | undefined;
+    } catch {
+      // not sure if this is what I should be returning here
+      return undefined;
+    }
   }
 
-  async delete(id: string) {
+  async updateAttributes(id: string, patch: Partial<TData>): Promise<TData> {
+    // this is kind of a different approach than the original UserData
+    // just more error handling
+    const current = await this.readOne(id);
+    if (!current) {
+      // will throwing Errors here be bubbled up properly?
+      // or should I return undefined or null?
+      throw new Error(`Record not found for id ${id}`);
+    }
+    const updated = { ...current, ...patch };
+    await this.userData.write(id, updated as z.input<TSchema>);
+    const after = await this.readOne(id);
+    if (!after) {
+      throw new Error(`Failed to update record for id ${id}`);
```

Comment:
Similarly we usually don't throw from these when write failed. Generally speaking I'd suggest to align with the current behavior of those without changing it